### ISSUE TITLE
编译成二进制后，怎么刷机

### ISSUE THREAD

**Issue by depfish** [tkso1997/openwrt-ac2100#2]
请问大佬，我编译成二进制后，AC2100怎么刷机呢！

有刷成功过麽。我小白，刚买，没刷机经验，担心变砖头。

多谢！

**Comment by hnjxlsj** [comment]
哇塞，你这自称“小白”，不名副其实啊。。还编译二进制了已经。。

**Comment by depfish** [comment]
> 哇塞，你这自称“小白”，不名副其实啊。。还编译二进制了已经。。

哪位大佬 会刷机啊。编译成了upgrade 固件，但不知道怎么刷，太尴尬了。这个Redmi AC2100没有开放root。

**Comment by hnjxlsj** [comment]
你应该说的是SSH吧。需要拆机，用编程线

**Comment by modzilla99** [comment]
Are you talking about flashing the firmware on the Redmi AC2100? Can you access UART? What is supported right now?

**Comment by hnjxlsj** [comment]
> Are you talking about flashing the firmware on the Redmi AC2100? Can you access UART? What is supported right now?

Yes, we are really talking about it.Up to now,  The owner of this post said that he/she has compiled  binary upgrade firmware.But we have no method to flash it because XiaoMi doesn't support SSH for us users.

**Comment by ko2001** [comment]
So far its only possible by manually flashing the NAND, which requires desoldering of the chip flashing firmware with boot wait enabled and resoldering it back. Afterwards you can flash the openwrt firmware via tftp and serial connection.

**Comment by hnjxlsj** [comment]
> Are you talking about flashing the firmware on the Redmi AC2100? Can you access UART? What is supported right now?

To be honest,I don't know how to flash a firmware on Redmi AC2100.I'm here to find good ideas that you may share.But I can serve as a translater here. Ahaha……

**Comment by hnjxlsj** [comment]
> So far its only possible by manually flashing the NAND, which requires desoldering of the chip flashing firmware with boot wait enabled and resoldering it back. Afterwards you can flash the openwrt firmware via tftp and serial connection.

I think you are right.Many users said as what you said.It requires more skills,which I may master for quite a long time！I am just a senior student comes from China,therefore my English is not too good.Last,it's 00:27 in China and I'd gonna sleep..........

**Comment by ko2001** [comment]
AFAIK there is people in the openwrt forum trying to find weak points in the firmware to find ways of flashing without messing with the hardware, but so far no success. 

Flashing the firmware is currently very difficult and requires good soldering skills, so its not usable for the mass.

**Comment by depfish** [comment]
> AFAIK there is people in the openwrt forum trying to find weak points in the firmware to find ways of flashing without messing with the hardware, but so far no success.
> 
> Flashing the firmware is currently very difficult and requires good soldering skills, so its not usable for the mass.

thanks all of you guys, I just follow your advices and did some find out. Unfortunately, currently there is only way to get SSH/root access via desoldering of the chip then flashing the NAND flash.

It's too difficulty for me, now I'm no idea to do that.  Maybe, I can try in future. 

Hopefully, some hardware hacker/master can provide us an easy way.

Thanks again:)

**Comment by modzilla99** [comment]
As I don't own the router (yet maybe), I'm not sure if the chip is accessible enough, but in alot of cases you're able to flash the chip without desoldering it. You just need a cheap CH341A for example with a clamp. Just align it on the NAND and make a backup with flashrom or any other software of your choice. Do it at least twice or more and check if you've got a good read! It's the only way back if you mess it up! With a hex editor you can then change to boot delay of U-Boot to 10 seconds so that you can get a serial console and enable ssh etc. Maybe you can enable ssh with some hex edit as well, I'm not sure exactly. But I would be happy to assist!


I'm not native as well, I'm German. ;)

**Comment by depfish** [comment]
> As I don't own the router (yet maybe), I'm not sure if the chip is accessible enough, but in alot of cases you're able to flash the chip without desoldering it. You just need a cheap CH341A for example with a clamp. Just align it on the NAND and make a backup with flashrom or any other software of your choice. Do it at least twice or more and check if you've got a good read! It's the only way back if you mess it up! With a hex editor you can then change to boot delay of U-Boot to 10 seconds so that you can get a serial console and enable ssh etc. Maybe you can enable ssh with some hex edit as well, I'm not sure exactly. But I would be happy to assist!
> 
> I'm not native as well, I'm German. ;)

It seems,  I  remember  someone post  like what you say in  the forum. Maybe I can try it.

This method much easier than desoldering.

**Comment by modzilla99** [comment]
It's buried somewhere in this thread: https://forum.openwrt.org/t/xiaomi-mi-router-4a-gigabit-edition-r4ag-r4a-gigabit-fully-supported-but-requires-overwriting-spi-flash-with-programmer/36685

**Comment by depfish** [comment]
![1585988445118](https://user-images.githubusercontent.com/28501184/78422237-6c25b480-7690-11ea-83fa-4e76cc9a9903.jpg)

@modzilla99 I tested, it's not working on AC2100.
